### PR TITLE
split out benchmark into own function

### LINKED ISSUE #69: Slowdown in bench_pairwise and bench_colwise for Julia v0.6
Running the benchmark tools in test/ shows the the basic looping code (*not* the colwise and pairwise code) is much slower (3-4x) in Julia 0.6 than it was in 0.5... does anyone know what's going on?

**v.0.5**
|  distance  |  loop  |  colwise  |  gain  |
|----------- | -------| ----------| -------|
| SqEuclidean | 0.007357s |  0.001989s |  3.6983 |
| Euclidean | 0.007447s |  0.002041s |  3.6488 |
| Cityblock | 0.007344s |  0.001986s |  3.6977 |
...

**v0.6**
|  distance  |  loop  |  colwise  |  gain  |
|----------- | -------| ----------| -------|
| SqEuclidean | 0.027896s |  0.001776s | 15.7099 |
| Euclidean | 0.029269s |  0.001811s | 16.1622 |
| Cityblock | 0.029836s |  0.001762s | 16.9368 |
...

### PR DESCRIPTION
**Pull request by KristofferC**
Try to fix #69. Can you try this out @richardreeve

### CODE REVIEW

File: test/bench_colwise.jl [hunk 1]
```diff
@@ -6,18 +6,20 @@ using BenchmarkTools
 
 BenchmarkTools.DEFAULT_PARAMETERS.seconds = 1.0
 
+function evaluate_colwise{T}(::Type{T}, dist, x, y)
+    n = size(x, 2)
+    r = Vector{T}(n)
+    for j = 1:n
```
Comment (ararslan) [inline test/bench_colwise.jl]: Would it help at all to `@inbounds` this? (And below)

Comment (KristofferC) [inline test/bench_colwise.jl]: The slice below should completely dominate a boundscheck.

**Comment by richardreeve** [comment]
@KristofferC Yes that seems to have completely fixed it:

*v0.5*
```
|  distance  |  loop  |  colwise  |  gain  |
|----------- | -------| ----------| -------|
| SqEuclidean | 0.005814s |  0.002057s |  2.8261 |
| Euclidean | 0.005815s |  0.002051s |  2.8353 |
| Cityblock | 0.005786s |  0.002004s |  2.8869 |
| Chebyshev | 0.009731s |  0.005480s |  1.7759 |
...
```

*v0.6*
```
|  distance  |  loop  |  colwise  |  gain  |
|----------- | -------| ----------| -------|
| SqEuclidean | 0.005913s |  0.001830s |  3.2306 |
| Euclidean | 0.005680s |  0.001817s |  3.1263 |
| Cityblock | 0.005695s |  0.001800s |  3.1639 |
| Chebyshev | 0.009120s |  0.004631s |  1.9696 |
...
```

Though it's going to conflict with #68 assuming you merge it as I'm adding new benchmarks too...

**Comment by KristofferC** [comment]
Perhaps you can cherry-pick this commit (or copy paste depending on your git skillz)

**Comment by ararslan** [comment]
Shouldn't be difficult to rebase out your commits that modify the benchmark times and resolve conflicts in the code manually.

**Comment by richardreeve** [comment]
@KristofferC Now my code has merged, the benchmarking includes RenyiDivergence as well, so you'll have to rerun it to include those stats. Also note that I run four different RenyiDivergence cases for speed as there are four different cases in the code depending on the parameter values. I didn't change the benchmarking code to pick up the parameter though, so I fixed it by hand to show which case is which (they are 0, 1, 2, ∞).

File: README.md [hunk 2]
```diff
@@ -185,70 +185,62 @@ julia> pairwise(Euclidean(1e-12), x, x)
...
+| Euclidean | 0.011099s |  0.003700s |  2.9995 |
+| Cityblock | 0.010886s |  0.003685s |  2.9545 |
+| Chebyshev | 0.013489s |  0.006112s |  2.2069 |
+| Minkowski | 0.125494s |  0.113365s |  1.1070 |
+| Hamming | 0.011138s |  0.003617s |  3.0791 |
+| CosineDist | 0.013189s |  0.004852s |  2.7181 |
+| CorrDist | 0.021433s |  0.017582s |  1.2190 |
+| ChiSqDist | 0.013495s |  0.004730s |  2.8532 |
+| KLDivergence | 0.045066s |  0.034814s |  1.2945 |
+| JSDivergence | 0.088533s |  0.078719s |  1.1247 |
```
Comment (richardreeve) [inline README.md]: RenyiDivergence(0) - RenyiDivergence(∞) missing here.

File: README.md [hunk 3]
```diff
@@ -185,70 +185,62 @@ julia> pairwise(Euclidean(1e-12), x, x)
...
+| Euclidean | 0.024351s |  0.000429s | **56.7555** |
+| Cityblock | 0.024320s |  0.001902s | 12.7882 |
+| Chebyshev | 0.029328s |  0.011291s |  2.5974 |
+| Minkowski | 0.327542s |  0.285311s |  1.1480 |
+| Hamming | 0.023939s |  0.001834s | 13.0540 |
+| CosineDist | 0.028258s |  0.000328s | **86.2659** |
+| CorrDist | 0.049505s |  0.000444s | **111.6189** |
+| ChiSqDist | 0.028372s |  0.010767s |  2.6352 |
+| KLDivergence | 0.105202s |  0.091300s |  1.1523 |
+| JSDivergence | 0.217159s |  0.197847s |  1.0976 |
```
Comment (richardreeve) [inline README.md]: RenyiDivergence(0) - RenyiDivergence(∞) also missing here.

**Comment by KristofferC** [comment]
Aha thanks for telling me. I'll try to rerun them at some point today.

**Comment by KristofferC** [comment]
Remade this into using a more modern way of writing the benchmark (define a suite, tune it, run it etc). Also ran the benchmarks.

**Pull request merged**